proposals/features-composite-dependencies.md

```diff
@@ -0,0 +1,117 @@
+# Composite Features
+
+Related: https://github.com/devcontainers/spec/issues/109
+
+## Motivation
+
+We've seen significant interest in the ability to "reuse" or "extend" a given Feature with one or more additional Features.  Often in software a given tool depends on another (or several) tool(s)/framework(s).  As the dev container Feature ecosystem has grown, there has been a growing need to reduce redundant code in Features by utilizing first installing a set of dependant Features.
+
```

Comment:
Would we want to link to https://github.com/devcontainers/spec/pull/209 as well, since both are related to sharing or reusing Feature code in some way?